util/src/main/java/tc/oc/pgm/util/nms/v1_10_12/NMSHacks1_10_12.java

```diff
@@ -0,0 +1,155 @@
+package tc.oc.pgm.util.nms.v1_10_12;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.SetMultimap;
+import java.util.Collection;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.Executors;
+import org.bukkit.Material;
+import org.bukkit.Server;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.material.MaterialData;
+import org.bukkit.plugin.Plugin;
+import tc.oc.pgm.util.attribute.AttributeMap;
+import tc.oc.pgm.util.attribute.AttributeModifier;
+import tc.oc.pgm.util.nms.attribute.AttributeMapBukkit;
+import tc.oc.pgm.util.nms.reflect.Refl;
+import tc.oc.pgm.util.nms.reflect.ReflectionProxy;
+
+public class NMSHacks1_10_12 extends NMSHacksProtocolLib1_10_12 {
+  static Refl refl = ReflectionProxy.getProxy(Refl.class);
+  static Refl.NBTTagString nbtTagString = ReflectionProxy.getProxy(Refl.NBTTagString.class);
+  static Refl.NBTTagList nbtTagList = ReflectionProxy.getProxy(Refl.NBTTagList.class);
+  static Refl.NBTTagCompound nbtTagCompound = ReflectionProxy.getProxy(Refl.NBTTagCompound.class);
+  static Refl.CraftMagicNumbers craftMagicNumbers =
+      ReflectionProxy.getProxy(Refl.CraftMagicNumbers.class);
+  static Refl.Block reflBlock = ReflectionProxy.getProxy(Refl.Block.class);
+  static Refl.IBlockData reflIBlockData = ReflectionProxy.getProxy(Refl.IBlockData.class);
+  static Refl.CraftItemStack craftItemStack = ReflectionProxy.getProxy(Refl.CraftItemStack.class);
+
+  @Override
+  public Set<Material> getMaterialCollection(ItemMeta itemMeta, String key) {
+    Map<String, Object> unhandledTags = refl.getUnhandledTags(itemMeta);
+    if (!unhandledTags.containsKey(key)) return EnumSet.noneOf(Material.class);
+    EnumSet<Material> materialSet = EnumSet.noneOf(Material.class);
+    Object canDestroyList = unhandledTags.get(key);
+
+    for (Object item : (List<Object>) nbtTagList.getListField(canDestroyList)) {
+      String blockString = nbtTagString.getString(item);
+      Object nmsBlock = reflBlock.getBlockByName(blockString);
+      int id = reflBlock.getId(nmsBlock);
+      Material material = Material.getMaterial(id);
+      materialSet.add(material);
+    }
+
+    return materialSet;
+  }
```

Comment:
this is an identical copy-paste to the one in 1.9 nms hacks

The point of reflection is that the same code can operate in multiple underlying types, this should be no different. This piece of code should not be replicated for each version, only the reflection part should use different methods

Reply:
Yeah, I was unsure how to structure the inheritance.

I was trying to keep reflection out of the protocol lib NMSHacks, because it was breaking things with the old static initializations, but given the new reflection annotations that should no longer be necessary. I'll consolidate the versioned NMSHacks since the separation should no longer be necessary.